crypto/s2n_rsa_pss.c

```diff
@@ -183,7 +183,7 @@ static int s2n_rsa_pss_key_free(struct s2n_pkey *pkey)
 }
 
 int s2n_evp_pkey_to_rsa_pss_public_key(struct s2n_rsa_key *rsa_key, EVP_PKEY *pkey) {
-    RSA *pub_rsa_key = EVP_PKEY_get0_RSA(pkey);
+    RSA *pub_rsa_key = EVP_PKEY_get1_RSA(pkey);
```

Comment:
What frees the extra reference this is adding?
How is this preferable to ```RSA *priv_rsa_key = (RSA *)EVP_PKEY_get0_RSA(pkey);```

Reply:
I would really hope the compiler would scream about that xD Let's try to avoid casting to remove the const. That's just hiding the problem, and could lead to bugs if we actually try to modify the key.

Reply:
> That's just hiding the problem
yea I dont like casting since that simply hides the issue. more discussion on this below.

Reply:
If we switch to `EVP_PKEY_get0_RSA` then we shouldnt be calling free: https://github.com/aws/s2n-tls/pull/3459#discussion_r953242055

https://www.openssl.org/docs/manmaster/man3/EVP_PKEY_get0_RSA.html

> ... EVP_PKEY_get0_RSA(), ... return the referenced key in pkey or NULL if the key is not of the correct type. The reference count of the returned key is not incremented and so the key must not be freed after use.